File: EducNotes.API/Models/LessonContent.cs
namespace EducNotes.API.Models
{
    public class LessonContent
    {
        public int Id { get; set; }
        public string Name { get; set; }
        public string Desc { get; set; }
        public int LessonId { get; set; }
        public Lesson Lesson { get; set; }
        public int NbHours { get; set; }
        public byte? SessionNum { get; set; }
    }
}

File: EducNotes-SPA/src/app/_models/lesson-content.ts
// import {} from 'class-validator';
import { Lesson } from './lesson';

export class LessonContent {
  public id: number;

  public name: string;

  public desc: string;

  public lessonId: number;

  public lesson: Lesson;

  public nbHours: number;

  public sessionNum: number;

  constructor(
    id?: number,
    name?: string,
    desc?: string,
    lessonId?: number,
    lesson?: Lesson,
    nbHours?: number,
    sessionNum?: number,
    ) {
    this.id = id || 0;
    this.name = name || '';
    this.desc = desc || '';
    this.lessonId = lessonId || 0;
    this.nbHours = nbHours || 0;
    this.sessionNum = sessionNum || null;
  }
}
